### PR TITLE
SNS - Accelerometer REPL Commands

### PR DESCRIPTION
**Pull request by licornes-fluos**
As we are using the same accelerometer code as last year, here is the code to test it with the current accelerometer.

### CODE REVIEW

File: lib/debug/commands/accelerometer_commands.cpp [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+  const auto optional_i2c = repl->getI2c(bus);
+  if (!optional_i2c) {
+    logger.log(core::LogLevel::kFatal, "Error creating I2C bus");
+    return core::Result::kFailure;
+  };
+  const auto i2c = std::move(*optional_i2c);
+
+  // instance of sensor
+  const auto accelerometer_sensor           = std::make_shared<sensors::Accelerometer>(logger, i2c);
+  const auto accelerometer_read             = "read";
```
Comment (TomLonergan03) [inline lib/debug/commands/accelerometer_commands.cpp:28]: var name `accelerometer_read_name` and set the value `"accelerometer read"`

File: lib/debug/commands/accelerometer_commands.cpp [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+  const auto accelerometer_read_description = "Read accelerometer";
+  const auto accelerometer_read_handler     = [&logger, accelerometer_sensor]() {
+    const auto value_ready  = accelerometer_sensor->isValueReady();  // method already returns state
```
Comment (TomLonergan03) [inline lib/debug/commands/accelerometer_commands.cpp:31]: this should be used, also what does the comment mean?

File: lib/debug/commands/accelerometer_commands.cpp [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+    const auto acceleration = accelerometer_sensor->read();
+    if (!acceleration) {
+      logger.log(core::LogLevel::kFatal, "Failed to read acceleration");
+      return;
+    }
+    logger.log(core::LogLevel::kDebug, "Acceleration: %d", *acceleration);
```
Comment (TomLonergan03) [inline lib/debug/commands/accelerometer_commands.cpp:37]: `*acceleration` is a struct, i don't think you can print it like that

**Review by TomLonergan03 — CHANGES_REQUESTED** [review]

File: lib/debug/commands/accelerometer_commands.cpp [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+  auto accelerometer_sensor           = std::move(*optional_accelerometer_sensor);
+  const auto read_command_name        = "accelerometer read";
+  const auto read_command_description = "Read from the accelerometer";
+  const auto read_command_handler     = [&logger, accelerometer_sensor]() {
+    const auto value_ready = accelerometer_sensor.isValueReady();
+    if (!value_ready) {
+      logger.log(core::LogLevel::kFatal, "Value is not ready");
+      return;
+    }
+    const auto acceleration = accelerometer_sensor.read();
```
Comment (licornes-fluos) [inline lib/debug/commands/accelerometer_commands.cpp:50]: How to I call this? I tried multiple ways, but it doesn't work. (same for value_ready above)

File: lib/sensors/accelerometer.hpp [hunk 5]
```diff
@@ -17,16 +17,15 @@ namespace hyped::sensors {
...
+enum class accelerometerAddress { k1D = 0x1D, k1E = 0x1E };
 
 class Accelerometer : public II2cMuxSensor<core::RawAccelerationData> {
  public:
   static std::optional<Accelerometer> create(core::ILogger &logger,
                                              std::shared_ptr<io::II2c> i2c,
                                              const std::uint8_t channel,
-                                             const std::uint8_t device_address);
+                                             const accelerometerAddress device_address);
+  //  const std::uint8_t device_address);
```
Comment (davidbeechey) [inline lib/sensors/accelerometer.hpp:28]:
```suggestion
```

**Review by davidbeechey — APPROVED** [review]

**Review by davidbeechey — APPROVED** [review]

**Review by TomLonergan03 — APPROVED** [review]
lets gamble try merge

**Pull request merged**